corehq/apps/case_importer/suggested_fields.py

```diff
@@ -0,0 +1,88 @@
+from django.utils.translation import ugettext as _
+import itertools
+from corehq.apps.case_importer.util import get_case_properties_for_case_type, \
+    RESERVED_FIELDS
+from dimagi.ext import jsonobject
+
+
+def _combine_field_specs(field_specs, exclude_fields):
+    """
+    take a list of FieldSpec objects and return a sorted list where field is unique
+    and fields in exclude_fields are removed, and where the first mention of a
+    field in field_specs will win out over any repeats.
```

Comment:
Yeah it felt like it was going to be that simple, but we're doing a few things at the same time. Maybe there was a way to do this more clearly, but I went with normalizing the inputs in the outer function (so that the special fields and the dynamic fields were both lists of FieldSpecs, and exclude was a set) and then doing the merge/unique-by-field/first-wins/exclude/sort thing in an inner function with a doc string that said what all was going on. Trying to do them together in a couple list or dict comprehensions the way I started with just turned out to be more confusing than it needed to be.

If you were just saying I should rewrite this function to take two lists instead of one, and then do

```
combined_field_specs = {}
combined_field_specs.update({field_spec.field for field_spec in second_list})
combined_field_specs.update({field_spec.field for field_spec in first_list})
combined_field_specs = {key: value for key, value in combined_field_specs.items() if key not in exclude_fields}
return sorted(combined_field_specs.values(), key=lambda field_spec: field_spec.field)
```

I could do that, but honestly I'm not sure it's that much better one way or the other.